lib/src/back_end/code_writer.dart

```diff
@@ -59,16 +59,29 @@ class CodeWriter {
   /// The options for the current innermost piece being formatted.
   _PieceOptions get _options => _pieceOptions.last;
 
+  /// The offset in the formatted code where the selection starts, or `null` if
```

Comment:
This pair of fields makes sense when either can be null independently. These comments make it seem like if one is non-null the other will be as well. Consider changing the docs to say what it means when one is null but not the other.

Reply:
Done!